src/internal/ppsutil/util.go

```diff
@@ -42,18 +42,32 @@ import (
 	ppsServer "github.com/pachyderm/pachyderm/v2/src/server/pps"
 )
 
+// MaxPipelineNameLength is the maximum allowed length of a pipeline name.  This
+// is based on the maximum length of a label in k8s is 63, and we have some
+// boilerplate which eats up part of that.  The tricky part is that the version
+// is monotonically increasing.  The workaround is to reserve enough space for
+// 9,999 pipeline versions, then truncate the pipeline name itself thereafter.
+var MaxPipelineNameLength = 63 - len("pipeline--v1234")
+
 // PipelineRepo creates a pfs repo for a given pipeline.
 func PipelineRepo(pipeline *pps.Pipeline) *pfs.Repo {
 	return client.NewRepo(pipeline.Name)
 }
 
 // PipelineRcName generates the name of the k8s replication controller that
-// manages a pipeline's workers
+// manages a pipeline's workers.  Names are limited to 63 characters, so the
+// pipeline name is truncated if necessary in order to stay under the limit.  Of
+// note, collisions are _not_ currently handled.
 func PipelineRcName(name string, version uint64) string {
 	// k8s won't allow RC names that contain upper-case letters
 	// or underscores
 	// TODO: deal with name collision
 	name = strings.ReplaceAll(name, "_", "-")
+	var max = 63 - (len("pipeline") + 1) // pipeline-
+	max -= len(fmt.Sprint(version)) + 2  // -v$VERSION
```

Comment:
I feel like you might as well just use `len("pipeline-")` and `fmt.Sprint("-v%d",version)`, or maybe even a single line, rather than adding the fixed characters as numbers separately, makes it easier to see that things match up